extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -107,18 +104,44 @@ export async function runRemoteQuery(cliServer: cli.CodeQLCliServer, credentials
     return; // No error message needed, since `getRepositories` already displays one.
   }
 
-  await runRemoteQueriesApiRequest(credentials, ref, language, repositories, query);
+  // Get the controller repo from the config, if it exists.
+  // If it doesn't exist, prompt the user to enter it, and save that value to the config.
+  let controllerRepo: string | undefined;
+  controllerRepo = getRemoteControllerRepo();
+  if (!controllerRepo) {
+    void logger.log('No controller repository defined.');
+    controllerRepo = await window.showInputBox({
+      title: 'Controller repository in which to display progress and results of remote queries',
+      placeHolder: '<owner>/<repo>',
+      prompt: 'Enter the name of a GitHub repository in the format <owner>/<repo>',
+      ignoreFocusOut: true,
+    });
+    if (!controllerRepo) {
+      void showAndLogErrorMessage('No controller repository entered.');
+      return;
+    } else if (!REPO_REGEX.test(controllerRepo)) { // Check if user entered invalid input
+      void showAndLogErrorMessage('Invalid repository format. Must be a valid GitHub repository in the format <owner>/<repo>.');
+      return;
+    }
+    void logger.log(`Setting the controller repository as: ${controllerRepo}`);
+    await setRemoteControllerRepo(controllerRepo);
```

Comment:
I added this to avoid users having to re-enter the controller repo each time. Not sure if this is the right thing to do, but I don't expect people will need to switch between controller repos regularly...

Reply:
What I think this does (esp now that I've pushed a small fix):

- If there's a valid repo in the `codeQL.remoteQueries.controllerRepo` setting, use that
- If not, get the user to enter it in a text box, and set that as `codeQL.remoteQueries.controllerRepo` (so they don't have to re-enter it next time)

Reply:
Ah...that makes sense now. Thanks.